src/__tests__/validator_arrayOf_tests.js

```diff
@@ -0,0 +1,70 @@
+import test from 'ava';
+
+import Validator from '../validator';
+
+test('Validator.validate should flatten structure if arrayOf rule is used', t => {
+  const validatorConfig = {
+    fields: {
+      arrayOf: {
+        key: { numeric: { integerOnly: true, max: 32 } },
+        value: { required: true, length: { max: 5 } },
+      },
+    },
+  };
+
+  const validationOutput = new Validator(validatorConfig)
+    .validate(['fields'], { fields: [{ key: 33, value: '1234561' }] });
+
+  const expectedOutput = {
+    'fields[0].key': {
+      field: 'fields[0].key',
+      rule: 'numeric.max',
+      value: 33,
+      config: { numeric: { integerOnly: true, max: 32 } },
+    },
+    'fields[0].value': {
+      field: 'fields[0].value',
+      rule: 'length.max',
+      value: '1234561',
+      config: { required: true, length: { max: 5 } },
+    },
+  };
+
+  t.deepEqual(expectedOutput, validationOutput);
+});
+
+test('Validator.validate should support nesting of arrayOf rule', t => {
```

Comment:
I think `'Validator.validate should support nesting of rules that return arrays of errors'`, or something similar, better conveys the intent of the test since we want it to work for other rules than arrayOf.